File: src/ProjectR/content/scripts/afflictions/Passives/JackOfAllTrades.lua
function GetName()
  return "Jack Of All Trades"
end

function GetType()
  return Passive
end

function GetHookPoints()
  return { TurnTriggered, BuffingStat }
end

function OnAttachment(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint == TurnTriggered then
    char:UseMP(-10)
  end

  if hookPoint == BuffingStat then
    param2 = param2 * 1.5
  end
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/JackOfAllTrades.cs
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Afflictions
{
    public class JackOfAllTrades : Affliction
    {
        public override string Name { get { return "Jack Of All Trades"; } }

        public override AfflictionType Type { get { return AfflictionType.Passive; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.TurnTriggered, HookPoint.BuffingStat, 
                };
            }
        }

        protected override void OnTurnTriggered(ICharacter character)
        {
            character.UseMP(-10);
        }

        protected override void OnBuffingStat(ICharacter character, ref Stat stat, ref double value)
        {
            if (value > 0)
            {
                value *= 1.5;
            }
        }
    }
}